File: test-utilities/src/main/scala/com/taskadapter/connector/testlib/FieldRowBuilder.scala
package com.taskadapter.connector.testlib

import com.taskadapter.connector.FieldRow
import com.taskadapter.model.Field

object FieldRowBuilder {

  def rows(field: Seq[Field[_]]): Seq[FieldRow[_]] = {
    field.map(f => build(f))
  }

  def build[T](f: Field[T]): FieldRow[T] = new FieldRow[T](Option(f), Option(f), null.asInstanceOf[String])
}

File: test-utilities/src/main/java/com/taskadapter/connector/testlib/FieldRowBuilder.java
package com.taskadapter.connector.testlib;

import com.taskadapter.connector.FieldRow;
import com.taskadapter.model.Field;

import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

public class FieldRowBuilder {
    public static List<FieldRow<?>> rows(List<Field<?>> fields) {
        return fields.stream()
                .map(FieldRowBuilder::build)
                .collect(Collectors.toList());
    }

    private static <T> FieldRow<T> build(Field<T> f) {
        return new FieldRow<T>(Optional.of(f), Optional.of(f), null);
    }
}
